prow/pubsub/subscriber/subscriber.go

```diff
@@ -212,7 +213,7 @@ func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEven
 
 	presubmits := cfg.GetPresubmitsStatic(orgRepo)
 	if prh.GitClient != nil { // Get from inrepoconfig only when GitClient is provided
-		presubmitsWithInrepoconfig, err := cfg.GetPresubmits(prh.GitClient, orgRepo, baseSHAGetter, headSHAGetters...)
+		presubmitsWithInrepoconfig, _, _, err := cache.GetPresubmitsFromCache(cfg.GetPresubmits, prh.GitClient, orgRepo, baseSHAGetter, headSHAGetters...)
```

Comment:
> So even if we just cache the ProwYAML, it will still be possible for the cache result to be stale if the merge method is changed.

Why? Even if the merge strategy is changed, if the baseSHA and headSHAs stay the same, the end result (what the git tree looks like post-merge) should be the same, right?

Reply:
I tried to move the defaulting stuff out of `defaultProwYAMLGetter` with a naive change but it broke a number of existing unit tests. Looks like the refactoring of `defaultProwYAMLGetter` to reuse the defaulting logic is tricker than I initially expected. I think this is the only blocker for this PR now.

Reply:
Done. PTAL